build.sh

```diff
@@ -6,104 +6,128 @@ set -eu
 
 ## START STANDARD BUILD SCRIPT INCLUDE
 # adjust relative paths as necessary
-THIS_SCRIPT="$(greadlink -f "${BASH_SOURCE[0]}" 2>/dev/null || readlink -f "${BASH_SOURCE[0]}")"
-. "$(dirname "$THIS_SCRIPT")/resources/builder.inc.sh"
+THIS_SCRIPT="$(readlink -f "${BASH_SOURCE[0]}")"
+. "${THIS_SCRIPT%/*}/resources/builder.inc.sh"
 ## END STANDARD BUILD SCRIPT INCLUDE
 
 ################################ Main script ################################
 
+# Get the docker image ID
 function _get_docker_image_id() {
   echo "$(docker images -q s-keyman-website)"
 }
 
+# Get the Docker container ID
 function _get_docker_container_id() {
   echo "$(docker ps -a -q --filter ancestor=s-keyman-website)"
 }
 
 function _stop_docker_container() {
-  S_CONTAINER=$(_get_docker_container_id)
-  if [ ! -z "$S_CONTAINER" ]; then
-    docker container stop $S_CONTAINER
+  local S_KEYMAN_CONTAINER=$(_get_docker_container_id)
+  if [ ! -z "$S_KEYMAN_CONTAINER" ]; then
+    docker container stop $S_KEYMAN_CONTAINER
   else
     echo "No Docker container to stop"
   fi
 }
 
+function _delete_docker_image() {
+  local S_IMAGE=$(_get_docker_image_id)
+  if [ ! -z "$S_IMAGE" ]; then
+    builder_echo "Removing image $S_IMAGE for s.keyman.com"
+    docker rmi "$S_IMAGE"
+  else
+    builder_echo "No Docker s.keyman.com image to delete"
+  fi
+}
+
 builder_describe \
   "Setup s.keyman.com site to run via Docker." \
-  configure \
-  clean \
-  build \
-  start \
-  stop \
-  test \
+  "configure" \
+  "clean" \
+  "build" \
+  "start" \
+  "stop" \
+  "test" \
+  ":app          The s.keyman.com site" \
+  ":kmwversion   Generate the static file kmwversions.json"
 
 builder_parse "$@"
 
 # This script runs from its own folder
 cd "$REPO_ROOT"
 
-if builder_start_action configure; then
-  # Nothing to do
-  builder_finish_action success configure
-fi
+builder_run_action configure # no action
 
 if builder_start_action clean; then
+  # Cleanup static file
+  if [ -f ./metadata/kmwversions.json ]; then
+    rm ./metadata/kmwversions.json
+  fi
+
   # Stop and cleanup Docker containers and images used for the site
   _stop_docker_container
-
-  S_CONTAINER=$(_get_docker_container_id)
-  if [ ! -z "$S_CONTAINER" ]; then
-    docker container rm $S_CONTAINER
+  
+  S_KEYMAN_CONTAINER=$(_get_docker_container_id)
+  if [ ! -z "$S_KEYMAN_CONTAINER" ]; then
+    docker container rm $S_KEYMAN_CONTAINER
   else
     echo "No Docker container to clean"
   fi
-    
-  S_IMAGE=$(_get_docker_image_id)
-  if [ ! -z "$S_IMAGE" ]; then
-    docker rmi s-keyman-website
-  else 
-    echo "No Docker image to clean"
-  fi
-
+	
+  _delete_docker_image
+  
   builder_finish_action success clean
 fi
 
-if builder_start_action stop; then
-  # Stop the Docker container
-  _stop_docker_container
-  builder_finish_action success stop
-fi
 
+# Stop the Docker container
+builder_run_action stop _stop_docker_container
```

Comment:
This means this will be run on any action, including `configure`. Is that really what we want?

Reply:
I thought `builder_run_action` is just shorthand for a single-command action

Reply:
Yes, you are right. My bad. Sorry. I must have been tired...